### ISSUE TITLE
Poor results when detecting object with exported model (onnx, torchscript)

### ISSUE THREAD

**Issue by datdao1998** [ultralytics/yolov5#8772]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Detection

### Bug

I trained yolov5 model (lastest version) and exported model to onnx and torchscript.
My result with .pt model is extremely good, but with onnx and torchscript mode the results are quite poor.
This is an example
**a) Model pt** 
![image](https://user-images.githubusercontent.com/42546396/181675555-83171dcd-cab3-4a2a-a005-72556479989b.png)
**b) Model onnx** (same as **torchsript**)
![image](https://user-images.githubusercontent.com/42546396/181675736-5a0be53b-abef-4c35-a24d-4e2094a58b0d.png)
 

### Environment

- YOLO : Yolov5 + torch 1.11.0+cu113
- OS : Ubuntu 20.04
- Python: 3.8.13
- Datasets : PubTables-1M (https://github.com/microsoft/table-transformer)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@datdao1998 👋 Hello! Thanks for asking about YOLOv5 🚀 **benchmarks**. YOLOv5 inference is officially supported in 11 formats, and all formats are benchmarked for identical accuracy and to compare speed every 24 hours by the [YOLOv5 CI](https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml?query=event%3Aschedule).

💡 ProTip: Export to ONNX or OpenVINO for up to 3x CPU speedup. See [CPU Benchmarks](https://github.com/ultralytics/yolov5/pull/6613).
💡 ProTip: Export to TensorRT for up to 5x GPU speedup. See [GPU Benchmarks](https://github.com/ultralytics/yolov5/pull/6963).

Format                      | `export.py --include`         | Model
:---                        | --:                           | :--
[PyTorch](https://pytorch.org/)                     | -                             | `yolov5s.pt`
[TorchScript](https://pytorch.org/docs/stable/jit.html)                 | `torchscript`                 | `yolov5s.torchscript`
[ONNX](https://onnx.ai/)                        | `onnx`                        | `yolov5s.onnx`
[OpenVINO](https://docs.openvino.ai/latest/index.html)                    | `openvino`                    | `yolov5s_openvino_model/`
[TensorRT](https://developer.nvidia.com/tensorrt)                    | `engine`                      | `yolov5s.engine`
[CoreML](https://github.com/apple/coremltools)                      | `coreml`                      | `yolov5s.mlmodel`
[TensorFlow SavedModel](https://www.tensorflow.org/guide/saved_model)       | `saved_model`                 | `yolov5s_saved_model/`
[TensorFlow GraphDef](https://www.tensorflow.org/api_docs/python/tf/Graph)         | `pb`                          | `yolov5s.pb`
[TensorFlow Lite](https://www.tensorflow.org/lite)             | `tflite`                      | `yolov5s.tflite`
[TensorFlow Edge TPU](https://coral.ai/docs/edgetpu/models-intro/)         | `edgetpu`                     | `yolov5s_edgetpu.tflite`
[TensorFlow.js](https://www.tensorflow.org/js)               | `tfjs`                        | `yolov5s_web_model/`


## Benchmarks

Benchmarks below run on a Colab Pro with the YOLOv5 tutorial notebook <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a>. To reproduce:
```bash
python utils/benchmarks.py --weights yolov5s.pt --imgsz 640 --device 0
```

### Colab Pro V100 GPU

```
benchmarks: weights=/content/yolov5/yolov5s.pt, imgsz=640, batch_size=1, data=/content/yolov5/data/coco128.yaml, device=0, half=False, test=False
Checking setup...
YOLOv5 🚀 v6.1-135-g7926afc torch 1.10.0+cu111 CUDA:0 (Tesla V100-SXM2-16GB, 16160MiB)
Setup complete ✅ (8 CPUs, 51.0 GB RAM, 46.7/166.8 GB disk)

Benchmarks complete (458.07s)
                   Format  mAP@0.5:0.95  Inference time (ms)
0                 PyTorch        0.4623                10.19
1             TorchScript        0.4623                 6.85
2                    ONNX        0.4623                14.63
3                OpenVINO           NaN                  NaN
4                TensorRT        0.4617                 1.89
5                  CoreML           NaN                  NaN
6   TensorFlow SavedModel        0.4623                21.28
7     TensorFlow GraphDef        0.4623                21.22
8         TensorFlow Lite           NaN                  NaN
9     TensorFlow Edge TPU           NaN                  NaN
10          TensorFlow.js           NaN                  NaN
```

### Colab Pro CPU

```
benchmarks: weights=/content/yolov5/yolov5s.pt, imgsz=640, batch_size=1, data=/content/yolov5/data/coco128.yaml, device=cpu, half=False, test=False
Checking setup...
YOLOv5 🚀 v6.1-135-g7926afc torch 1.10.0+cu111 CPU
Setup complete ✅ (8 CPUs, 51.0 GB RAM, 41.5/166.8 GB disk)

Benchmarks complete (241.20s)
                   Format  mAP@0.5:0.95  Inference time (ms)
0                 PyTorch        0.4623               127.61
1             TorchScript        0.4623               131.23
2                    ONNX        0.4623                69.34
3                OpenVINO        0.4623                66.52
4                TensorRT           NaN                  NaN
5                  CoreML           NaN                  NaN
6   TensorFlow SavedModel        0.4623               123.79
7     TensorFlow GraphDef        0.4623               121.57
8         TensorFlow Lite        0.4623               316.61
9     TensorFlow Edge TPU           NaN                  NaN
10          TensorFlow.js           NaN                  NaN
```

Good luck 🍀 and let us know if you have any other questions!

**Comment by drishtirohan** [comment]
I am facing the same issue . The above image of bechmarks is same as mine but while using it for a model very poor results. I have trained my v5 model on 4000 images ,exported to saved_model format. 
Same happend when i used yolov5s.pt to export it to saved_model format the export showed the same image as above but while using , the model doesn't works good.

**Comment by pocca2048** [comment]
Similar to #8136, probably because the input is different.
The pytorch model takes rectangular inputs, but the exported model takes square inputs.
Accordingly, the latter includes large padding resulting performance falloff.

**Comment by glenn-jocher** [comment]
@pocca2048 thanks for reaching out! The issue you mentioned regarding the performance discrepancy between the PyTorch model and the exported model could be related to the input resolution difference.

To ensure consistent results, you can specify the input resolution when exporting the model using the `--img-size` argument with `export.py`. This will help ensure that the exported model is compatible with the input resolution used during training, potentially resolving the performance falloff you described.

Feel free to give this a try and let us know if it helps improve the model's performance!

And as always, the YOLOv5 community is here to help you with any further questions or concerns.
